cdap-securestore-spi/src/main/java/co/cask/cdap/securestore/spi/SecretAlreadyExistsException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.securestore.spi;
+
+/**
+ * Exception that indicates a secret already exists.
+ */
+public class SecretAlreadyExistsException extends Exception {
```

Comment:
I think Terence explained it very nicely as why this should be a checked exception. 

To add to that if the caller here is CDAP platform and the SPI is used in public APIs then with an unchecked exception the implementation can miss to catch it and send a user friendly response or http status code  (although in this case the status code will be correct due to our http exception handler which will send the right code due to the class name of the exception). The user will end up seeing SecretAlreadyExistsException exception  and its details which looks bad in the UI.

Reply:
This has been clarified and fixed already. So shall we consider it resolved then?

Reply:
Yes. As mentioned in the comment I was just adding to the point that why it should be checked.